zerver/data_import/microsoft_teams.py

```diff
@@ -25,9 +26,115 @@
 from zerver.models.recipients import Recipient
 from zerver.models.users import UserProfile
 
+
+@dataclass
+class TeamMetadata:
+    """
+    "team" is equivalent to a Zulip channel
+    """
+
+    description: str
+    display_name: str
+    visibility: Literal["public", "private"]
+    is_archived: bool
+    zulip_channel_id: int
+    zulip_recipient_id: int
+
+
+AddedTeamsT: TypeAlias = dict[str, TeamMetadata]
+TeamIdToZulipRecipientIdT: TypeAlias = dict[str, int]
 MicrosoftTeamsUserIdToZulipUserIdT: TypeAlias = dict[str, int]
 MicrosoftTeamsFieldsT: TypeAlias = dict[str, Any]
 
+MICROSOFT_TEAMS_DEFAULT_ANNOUNCEMENTS_CHANNEL_NAME = "All company"
+
+
+def convert_teams_to_channels(
+    microsoft_teams_user_id_to_zulip_user_id: MicrosoftTeamsUserIdToZulipUserIdT,
+    realm: dict[str, Any],
+    realm_id: int,
+    teams_data_dir: str,
+) -> AddedTeamsT:
+    team_data_folders = []
+    for f in os.listdir(teams_data_dir):
+        path = os.path.join(teams_data_dir, f)
+        if os.path.isdir(path):
+            team_data_folders.append(f)
+
+    logging.info("######### IMPORTING TEAMS STARTED #########\n")
+
+    # Build teams subscription map
+    team_id_to_zulip_subscriber_ids: dict[str, list[int]] = defaultdict(list)
```

Comment:
nitpick: mapping to a ` set` might be more robust than a set, in case we event visit the same `user_id` twice for a team?